### DISCUSSION TITLE
i can't reach my about page from url.

### DISCUSSION THREAD

**Discussion by hex909** [remix-run/react-router#9023] [Q&A] [answered]
hey i am a starter. I create this site with react and routing with react-router.  ''https://hex909countries.netlify.app/" I can access all pages in this website. but I can't access any page (path) from direct url or ''/country/Yemen" in this path if i refresh in this page it will shows 404 Page Not Found. how can i fix this. give me any resources that will helps me. 

and also not showing 404 page?

https://hex909countries.netlify.app/

here is my code https://github.com/hex909/countries

**Comment by bradleylandis** [comment, marked as answer]
Hello.  When creating a single page app, you need to tell your web server that all requests should actually be redirected to the single page.  I'm not very familiar with Netlify, but please see the "Avoid 404s" section here: https://docs.netlify.com/configure-builds/javascript-spas/#app

**Reply by hex909** [reply]
sorry for the delay. if there is any free hosting serves are there services

**Comment by mnaray** [comment]
For Netlify, you need to add a `_redirect` file to the public directory of your react-app. (_No file extension needed!_)

In that file you need to enter the following paths to allow every redirect.
`/* /index.html 200`
This should be the only line you need.

It basically tells the Netlify host to send the HTTP-code 200 (OK) to the client on every redirect.

I hope this helps. Please let me know if it doesn't! 😃

**Reply by hex909** [reply]
Thanks for the suggestion but that doesn't work for me. i got same error 'Page Not Found'

**Reply by mnaray** [reply]
Are you sure you published your deployment after building it?
Sometimes Netlify requires you to press a publish button on the build page.

**Reply by hex909** [reply]
first of all thanks for the interest.

yes i redeploy the build. and got some logs i dont know if this is useful. but i attached here
[netlify logs.txt](https://github.com/remix-run/react-router/files/9573593/netlify.logs.txt)

**Reply by mnaray** [reply]
I'll check out the logs as soon as I get to it.

Meanwhile you might find something interesting here regarding the routing with browser history.

https://create-react-app.dev/docs/deployment/

You have to scroll down a bit to see that.

**Reply by mnaray** [reply]
> first of all thanks for the interest.
> 
> yes i redeploy the build. and got some logs i dont know if this is useful. but i attached here
> [netlify logs.txt](https://github.com/remix-run/react-router/files/9573593/netlify.logs.txt)
> 
> 

The logs seem to be just fine as well.
I might try to run it locally, if I get to it.
Because if it doesn't work locally, then the issue probably is in your app's routing, not in netlify's redirect management.

**Reply by bradleylandis** [reply]
It looks like you have a bad redirect defined in https://github.com/hex909/countries/blob/main/public/netlify.toml  I'm not sure how it interacts if you put a _redirects file and a netlify.toml file.

**Reply by bradleylandis** [reply]
it looks like netlify.toml is supposed to be at the root of your repository:
https://docs.netlify.com/configure-builds/file-based-configuration/

**Reply by bradleylandis** [reply]
I would put this in the netlify.toml file and move it to the root of your repo and then delete the _redirect file:

```
# The following redirect is intended for use with most SPAs
# that handle routing internally.
[[redirects]]
  from = "/*"
  to = "/index.html"
  status = 200
```

**Reply by hex909** [reply]
Thanks for the solution @bradleylandis and supporters @mnaray

**Reply by mnaray** [reply]
Welcome!
